lib/matplotlib/axes/_axes.py

```diff
@@ -4995,13 +5002,37 @@ def fill_betweenx(self, y, x1, x2=0, where=None,
                 continue
 
             N = len(yslice)
-            Y = np.zeros((2 * N + 2, 2), float)
+            Y = np.zeros((2 * N + 2, 2), np.float)
+            if interpolate:
+                def get_interp_point(ind):
```

Comment:
I think so. I thought about that but also about using some interpolation function maintained in scipy (would this be a bad idea perhaps?). I'll have a look at it as soon as I can.